server/actions/voters.go

```diff
@@ -1 +1,71 @@
 package actions
+
+import (
+	"fmt"
+	"net/http"
+
+	database "durn/server/db"
+	"durn/server/util"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm/clause"
+)
+
+// AddVoters takes a list of email addresses and adds them all to the
+// database table `valid_voters`. It silently skips all strings that are
+// not valid email addresses and addresses that are already in the database
+func AddVoters(c *gin.Context) {
+	body := struct {
+		Voters []string `json:"voters" binding:"required"`
+	}{}
+
+	if !util.TryParseJsonBody(c, &body) {
+		return
+	}
+
+	var voters []database.ValidVoter
+	for _, voter := range body.Voters {
+		if util.ValidEmail(voter) {
+			voters = append(voters, database.ValidVoter{Email: voter})
+		}
+	}
+
+	db := database.GetDB()
+	defer database.ReleaseDB()
+
+	if err := db.Clauses(clause.OnConflict{DoNothing: true}).Create(&voters).Error; err != nil {
+		fmt.Println(err)
+		c.String(http.StatusInternalServerError, "Server failed to handle request")
+		return
+	}
+
+	c.String(http.StatusOK, "200 OK")
+}
+
+// RemoveVoters takes a list of email addresses and removes them from the database.
+// Ignores addresses that are not in the database
+func RemoveVoters(c *gin.Context) {
+	body := struct {
+		Voters []string `json:"voters" binding:"required"`
+	}{}
+
+	if !util.TryParseJsonBody(c, &body) {
+		return
+	}
```

Comment:
This pattern is used in multiple places. I am not good enough to say that this is not idiomatic Go. But to me, it feels very weird to catch an error in a util function and return T/F instead of the actual error and let the calling function handle it.

Reply:
I did it in a try to reduce code duplication, but I'm honestly inclined to agree, it feels a bit weird. It also has the disadvantage that it prevents writing clearer error messages if needed, so I'll probably revert.